// @expo-google-fonts/jetbrains-mono/index.js
// This file is generated by @expo-google-fonts/generator
// If you want to modify it, go to packages/generator/templates

export * from './useFonts';
export { default as __metadata__ } from './metadata.json';
export const JetBrainsMono_100Thin = require('./100Thin/JetBrainsMono_100Thin.ttf');
export const JetBrainsMono_200ExtraLight = require('./200ExtraLight/JetBrainsMono_200ExtraLight.ttf');
export const JetBrainsMono_300Light = require('./300Light/JetBrainsMono_300Light.ttf');
export const JetBrainsMono_400Regular = require('./400Regular/JetBrainsMono_400Regular.ttf');
export const JetBrainsMono_500Medium = require('./500Medium/JetBrainsMono_500Medium.ttf');
export const JetBrainsMono_600SemiBold = require('./600SemiBold/JetBrainsMono_600SemiBold.ttf');
export const JetBrainsMono_700Bold = require('./700Bold/JetBrainsMono_700Bold.ttf');
export const JetBrainsMono_800ExtraBold = require('./800ExtraBold/JetBrainsMono_800ExtraBold.ttf');
export const JetBrainsMono_100Thin_Italic = require('./100Thin_Italic/JetBrainsMono_100Thin_Italic.ttf');
export const JetBrainsMono_200ExtraLight_Italic = require('./200ExtraLight_Italic/JetBrainsMono_200ExtraLight_Italic.ttf');
export const JetBrainsMono_300Light_Italic = require('./300Light_Italic/JetBrainsMono_300Light_Italic.ttf');
export const JetBrainsMono_400Regular_Italic = require('./400Regular_Italic/JetBrainsMono_400Regular_Italic.ttf');
export const JetBrainsMono_500Medium_Italic = require('./500Medium_Italic/JetBrainsMono_500Medium_Italic.ttf');
export const JetBrainsMono_600SemiBold_Italic = require('./600SemiBold_Italic/JetBrainsMono_600SemiBold_Italic.ttf');
export const JetBrainsMono_700Bold_Italic = require('./700Bold_Italic/JetBrainsMono_700Bold_Italic.ttf');
export const JetBrainsMono_800ExtraBold_Italic = require('./800ExtraBold_Italic/JetBrainsMono_800ExtraBold_Italic.ttf');
